uuid/unstable_v6.ts

```diff
@@ -0,0 +1,178 @@
+
+import { bytesToUuid } from "./_common.ts";
+
+const UUID_RE =
+  /^[0-9a-f]{8}-[0-9a-f]{4}-6[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i;
+
+/**
+ * Determines whether a string is a valid
+ * {@link https://www.rfc-editor.org/rfc/rfc9562.html#name-uuid-version-6 | UUIDv6}.
+ *
+ * @param id UUID value.
+ *
+ * @returns `true` if the string is a valid UUIDv6, otherwise `false`.
+ *
+ * @example Usage
+ * ```ts
+ * import { validate } from "@std/uuid/unstable-v6";
+ * import { assert, assertFalse } from "@std/assert";
+ *
+ * assert(validate("1efed67d-d966-6490-8b9a-755015853480"));
+ * assertFalse(validate("1efed67d-d966-1490-8b9a-755015853480"));
+ * ```
+ */
+export function validate(id: string): boolean {
+  return UUID_RE.test(id);
+}
+
+let _nodeId: number[];
+let _clockseq: number;
+
+let _lastMSecs = 0;
+let _lastNSecs = 0;
+
+/** Options for {@linkcode generate}. */
+export interface GenerateOptions {
+  /**
+   * An array of 6 bytes that represents a 48-bit IEEE 802 MAC address.
+   *
+   * @see {@link https://www.rfc-editor.org/rfc/rfc4122#section-4.1.6}
+   */
+  node?: number[];
+  /**
+   * A 14-bit value used to avoid duplicates that could arise when the clock is
+   * set backwards in time or if the node ID changes (0 - 16383).
+   *
+   * @see {@link https://www.rfc-editor.org/rfc/rfc4122#section-4.1.5}
+   */
+  clockseq?: number;
+  /**
+   * The number of milliseconds since the Unix epoch (January 1, 1970).
+   *
+   * @see {@link https://www.rfc-editor.org/rfc/rfc4122#section-4.1.4}
+   */
+  msecs?: number;
+  /**
+   * The number of nanoseconds to add to {@linkcode GenerateOptions.msecs}
+   * (0 - 10,000).
+   *
+   * @see {@link https://www.rfc-editor.org/rfc/rfc4122#section-4.1.4}
+   */
+  nsecs?: number;
+  /** An array of 16 random bytes (0 - 255). */
+  random?: number[];
+  /**
+   * A function that returns an array of 16 random bytes (0 - 255).
+   * Alternative to {@linkcode GenerateOptions.random}.
+   */
+  rng?: () => number[];
+}
+
+/**
+ * Generates a
+ * {@link https://www.rfc-editor.org/rfc/rfc9562.html#name-uuid-version-6 | UUIDv6}.
+ *
+ * @param options Can use RFC time sequence values as overwrites.
+ *
+ * @returns Returns a UUIDv6 string.
+ *
+ * @example Usage
+ * ```ts
+ * import { generate, validate } from "@std/uuid/unstable-v6";
+ * import { assert } from "@std/assert";
+ *
+ * const options = {
+ *   node: [0x01, 0x23, 0x45, 0x67, 0x89, 0xab],
+ *   clockseq: 0x1234,
+ *   msecs: new Date("2011-11-01").getTime(),
+ *   nsecs: 5678,
+ * };
+ *
+ * const uuid = generate(options);
+ * assert(validate(uuid as string));
+ * ```
+ */
+export function generate(options: GenerateOptions = {}): string {
+  let i = 0;
+  const b: number[] = [];
+
+  let { node = _nodeId, clockseq = _clockseq } = options;
+
+  if (node === undefined || clockseq === undefined) {
+    // deno-lint-ignore no-explicit-any
+    const seedBytes: any = options.random ??
+      options.rng ??
```

Comment:
This seems needing to be called as function

Reply:
Can you add a test case to use `rng` option?

Reply:
Code adjusted and test introduced